### DISCUSSION TITLE
How to run isort locally and match results of github actions?

### DISCUSSION THREAD

**Discussion by dschult** [networkx/networkx#5722] [General]
I have isort installed via pip and it is version 5.10.1
When I run it, it changes the import order in my test file to

```python
...
import networkx as nx
import pytest
...
```
But when I push that to the PR, isort is run via github actions and uses the arguments (based on the pre-commit.yaml file)
`isort --profile black --filter-files --skip __init__.py`
This complains that the order should be:
```python
...
import pytest

import networkx as nx
...
```
I believe this makes sense because while pytest and networkx are not builtin libraries, networkx is the library being constructed while pytest is a third party library we are using. It is impressive to me that isort knows to move networkx imports to the end... but how does it do that, how can I get it to do that locally, and how are we telling isort to do that on the github actions?

Thanks!

**Comment by MridulS** [comment]
>  When I run it, it changes the import order in my test file

How are you running `isort` locally?

If I create a file locally  (!!inside the networkx local repo!!, this behavior wouldn't work outside the repo as it will treat networkx as an "installed library")
``` python
...
import networkx as nx
import pytest
...
```
and run `$ isort filename.py` it is giving me the "correct" order.

`pre-commit` should already runs these lints for the user when they do a git commit.

We should be running `isort --profile black --filter-files --skip __init__.py networkx/` locally to get the exact same output as our CI if we want to do it manually without `pre-commit`.

**Reply by dschult** [reply]
Thanks @MridulS !
I am in the NetworkX repo locally on my machine. I am working on a PR created before the change that adds isort to pre-commit. So even with pre-commit installed for this repo, I don't get the isort before I commit.  I tried rebasing (it was over 1000 commits) and decided that was too much work.  So I installed isort and ran it locally:

```python
>>> isort --version

                 _                 _
                (_) ___  ___  _ __| |_
                | |/ _/ / _ \/ '__  _/
                | |\__ \/\_\/| |  | |_
                |_|\___/\___/\_/   \_/

      isort your imports, so you don't have to.

                    VERSION 5.10.1

>>> isort --diff tests/test_adigraph.py 
--- /Users/dschult/NXLuca/networkx/drawing/tests/test_adigraph.py:before	2022-06-10 13:01:07.592384
+++ /Users/dschult/NXLuca/networkx/drawing/tests/test_adigraph.py:after	2022-06-14 10:52:14.676380
@@ -1,9 +1,8 @@
 import filecmp
 import os
 
+import networkx as nx
 import pytest
-
-import networkx as nx
 
 
 def test_multiple_graphs():
```
This is the opposite of the change required by the linting system in the CI.  And I think that version of isort is also 5.10.1.  So I guess it must be the filters.  Or maybe because I am not calling it on the whole tree.  But when I do this from the home of the repo:
```
isort --profile black --filter-files --skip __init__.py networkx/
```
I get a very large number of the same changes (switching pytest and networkx imports) from lots of files. So I guess that is not it either.
